### PR TITLE
HS-621 remove core REST API endpoints in BFF 

### PR DESCRIPTION
**Pull request by yangli2014**
## Description

We can shut down the core now. The only API still pointing to the core is the data choice, however, the UI doesn't use it. 
We need to implement gRPC for alarms and the new data choice service.
Another thing is the notification is still using REST API, we need to switch to the gRPC as well.

## Jira link(s)
- https://issues.opennms.org/browse/HS-621

## Checklist
* [x] Follows Horizon Stream's [development guidelines.](https://github.com/OpenNMS/horizon-stream/wiki/Development-Guidelines)
* [x] Appropriate reviewer(s) have been selected.
* [x] Jira issue(s) have been updated to "In Review".
* [ ] Includes [appropriate tests.](https://github.com/OpenNMS/horizon-stream/wiki/Test-Strategy)
* [ ] Documentation has been updated as necessary.

### CODE REVIEW

File: charts/opennms/templates/opennms/minion/minion-configmap.yaml [hunk 1]
```diff
@@ -13,3 +13,17 @@ data:
   opennms-minion-identity: |
     id=${env:HOSTNAME:-UNKNOWN}
     location=${env:LOCATION:-Default}
+---
```
Comment (jkapponms) [inline charts/opennms/templates/opennms/minion/minion-configmap.yaml:16]: this needs to be in it's own file

Comment (artnaseef) [inline charts/opennms/templates/opennms/minion/minion-configmap.yaml:16]: The contents of the `opennms-config` can be merged with `minion-config` above to have a single config.

File: inventory/pom.xml [hunk 2]
```diff
@@ -62,11 +62,6 @@
             <version>${lombok.version}</version>
             <optional>true</optional>
         </dependency>
```
Comment (artnaseef) [inline inventory/pom.xml:64]: Did inventory change as part of removing core?

File: charts/opennms/templates/opennms/minion/minion-configmap.yaml (hunk 1 above)
Comment (yangli2014) [reply charts/opennms/templates/opennms/minion/minion-configmap.yaml:16]: ok, still in the minion folder?

File: rest-server/src/main/java/org/opennms/horizon/server/service/AlarmService.java [hunk 3]
```diff
@@ -1,63 +0,0 @@
-/*******************************************************************************
```
Comment (artnaseef) [inline rest-server/src/main/java/org/opennms/horizon/server/service/AlarmService.java:1]: Did Alarm service change as a part of removing core?

File: rest-server/src/main/java/org/opennms/horizon/server/service/DataChoicesService.java [hunk 4]
```diff
@@ -1,47 +0,0 @@
-package org.opennms.horizon.server.service;
```
Comment (artnaseef) [inline rest-server/src/main/java/org/opennms/horizon/server/service/DataChoicesService.java:1]: Is this change related to core?

File: rest-server/src/main/java/org/opennms/horizon/server/service/DeviceService.java [hunk 5]
```diff
@@ -1,66 +0,0 @@
-/*******************************************************************************
```
Comment (artnaseef) [inline rest-server/src/main/java/org/opennms/horizon/server/service/DeviceService.java:1]: Is this related to the removal of core?

File: rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlarmService.java [hunk 6]
```diff
@@ -28,34 +28,6 @@
 
 package org.opennms.horizon.server.service;
```
Comment (artnaseef) [inline rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlarmService.java:30]: Is this related to the removal of core?

Comment (yangli2014) [reply rest-server/src/main/java/org/opennms/horizon/server/service/GrpcAlarmService.java:30]: By removing the REST API-based service, it is a place holder for the gRPC service.

File: rest-server/src/main/java/org/opennms/horizon/server/service/AlarmService.java (hunk 3 above)
Comment (yangli2014) [reply rest-server/src/main/java/org/opennms/horizon/server/service/AlarmService.java:1]: yes

File: rest-server/src/main/java/org/opennms/horizon/server/service/DataChoicesService.java (hunk 4 above)
Comment (yangli2014) [reply rest-server/src/main/java/org/opennms/horizon/server/service/DataChoicesService.java:1]: yes

**Review by jkapponms — APPROVED** [review]

**Review by geraldhumphries — APPROVED** [review]
